### ISSUE TITLE
Sections defined within sections are finding sections out of scope (page level) on version 2.12 works as expected in version 2.9.

### ISSUE THREAD

**Issue by jdavis2007** [natritmeyer/site_prism#253]
# Issue to raise for SitePrism framework
Before raising an issue with us please ask yourself whether you have all the information available to advise us.

For us to try to help you with your issue, we require certain information. Please replace each of the bullet points with your **own** input.

## Environment
- Windows 10 Ruby 2.33

## Expected Behavior
When i define sections in a section the sections returned should be only section elements that are contained in the root element of the parent  section.

## Actual Behavior
The defined sections were returning 448 elements in the capybara array the amount  on my page in version 2.12 but only the expected 122 elements in my root element in version 2.9.

## Steps to reproduce
1) Given I have a page with multiple containers with each containing elements with the same selectors.  
2)  When I define a section that contains sections in version 2.12 if i run a test and debug the value of the defined sections in a watch it will collect all page elements matching the selector rather than just he sections inside the root element in the html.  If i use the same approach in version 2.9 it works as expected.

Working on creating an example page as I can not link to our project in which i found the issue.

**Comment by matisnape** [comment]
I'm also experiencing that. 
My stack:
- ruby 2.4.1
- siteprism 2.11

```
section :questions_section, :xpath, "//section[*[h3[contains(text(),'Questions')]]]" do
    sections :questions, :xpath, "//div[@class='question']" do
      element :label, :xpath, '//label'
      element :text_input, :xpath, "//input[@type='text']"
    end
  end

```
I wanted to check if one text_input is visible within each question section. When I call mapping on questions, I get ambigous match
```
[17] pry(#<RSpec::ExampleGroups::ScreeningQuestions::Adding::CanAddOpenTextQuestionSingleLine>)> careers_page.questions_section.questions.map { |item| item.text_input }
Capybara::Ambiguous: Ambiguous match, found 2 elements matching visible xpath "//input[@type='text']" within #<Capybara::Node::Element tag="div" path="/html/body/section/div/div/form/section[4]/div[2]/div[1]">
from /Users/anks/.rvm/gems/ruby-2.4.1/gems/capybara-2.18.0/lib/capybara/node/finders.rb:311:in `block in synced_resolve'
```

but when I check for the text_input with concatenated xpath, I get only one hit:
```
[18] pry(#<RSpec::ExampleGroups::ScreeningQuestions::Adding::CanAddOpenTextQuestionSingleLine>)> find(:xpath, "//section[*[h3[contains(text(),'Questions')]]]//div[@class='question']//input[@type='text']")
=> #<Capybara::Node::Element tag="input" path="/html/body/section/div/div/form/section[4]/div[2]/div[1]/div/input">```

In other words - works fine with simple find(:xpath), but doesn't with chaining the elements.

**Comment by matisnape** [comment]
@twalpole yeah, I confirm - after updating the xpaths, the section scoping works all right

**Comment by luke-hill** [comment]
To the OP. We did some fixes around scoping "not working" the way it should. And a release was cut (In 2.12), which massively changed the volume of results you would be getting returned.

Can you provide a reproducible script / sample page to look at. As this is something that is extensively tested, and the new features cover testing this in its entirety

**Comment by luke-hill** [comment]
Ping @jdavis2007 / @matisnape

**Comment by matisnape** [comment]
@luke-hill using xpaths with `.//` instead of `//` for elements inside sections solved my issue

**Comment by luke-hill** [comment]
Closing as fixed in v2.13 - If the issue still persists @jdavis2007 feel free to open a more detailed Issue report.
